src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolNewtonsoft.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline;
+
+/// <summary>
+/// These are very temporary types that we need in order to serialize document symbol data
+/// using Newtonsoft instead of System.Text.Json
+///
+/// We currently must support Newtonsoft serialization here because we have not yet opted into using STJ
+/// in the VS language server client (and so the client will serialize the request using Newtonsoft).
+/// 
+/// https://github.com/dotnet/roslyn/pull/72675 tracks opting in the client to STJ.
+/// TODO - everything in this type should be deleted once the client side is using STJ.
+/// </summary>
+internal class DocumentSymbolNewtonsoft
```

Comment:
I tried to contain the spread of newtonsoft here by just copying the types we need and adding the newtonsoft attributes to them.

These can be deleted as soon as we switch to STJ on the client side